src/backend/vulkan/mod.rs

```diff
@@ -532,6 +534,92 @@ impl PhysicalDevice {
         unsafe { instance.get_physical_device_properties2(self.handle(), props) }
     }
 
+    /// Get a list of supported formats.
+    ///
+    /// Checks a list of formats (currently hardcoded to be Argb8888 and Argb2101010 from DrmFourcc) to see if they are compatible with
+    /// the current device.
+    ///
+    /// Returns a vec of [`vk::Format`], with a length of 0 if none are found.
+    ///
+    /// ```
+    /// use smithay::{
+    ///     backend::vulkan::{version::Version, Instance, PhysicalDevice},
+    ///     reexports::ash::vk,
+    /// };
+    ///
+    /// let instance = Instance::new(Version::VERSION_1_3, None).unwrap();
+    ///
+    /// let physical_device = PhysicalDevice::enumerate(&instance)
+    ///     .unwrap()
+    ///     .next()
+    ///     .expect("No physical devices");
+    ///
+    /// let formats = physical_device.get_supported_formats();
+    ///
+    /// assert_eq!(
+    ///     formats,
+    ///     vec![
+    ///         vk::Format::B8G8R8A8_SRGB,
+    ///         vk::Format::A8B8G8R8_SRGB_PACK32,
+    ///         vk::Format::R8G8B8A8_SRGB,
+    ///         vk::Format::A2R10G10B10_UNORM_PACK32
+    ///     ]
+    /// );
+    /// ```
+    pub fn get_supported_formats(&self) -> Vec<vk::Format> {
```

Comment:
This is probably not the right spot for this specific function. But I'll explain where I'd like to see this later today. Specifically I want this inside of `VulkanRenderer` because this form of get_supported_formats is quite dependent on the image usage used by wl_shm.

Reply:
So what I'd like to see is this (the actual inner contents look fine):
1. Inside of `VulkanRenderer`, maintain a list of formats which are supported using logic inside this function. A `HashSet` should work fine there.
2. At the bottom of `VulkanRenderer::new`, loop over the hardcoded list of formats and if the format is supported using the logic you already have, add the format to the list. I'd try to split that initialization of the format list into a function to make sure VulkanRenderer::new doesn't grow too quickly.

Reply:
does this look good?